components/inputs/input-date.js

```diff
@@ -182,12 +185,19 @@ class InputDate extends SkeletonMixin(FormElementMixin(LocalizeCoreElement(LitEl
 
 		await (document.fonts ? document.fonts.ready : Promise.resolve());
 
-		const hiddenContent = this.shadowRoot.querySelector('.d2l-input-date-hidden-content');
+		const hiddenContent = this.shadowRoot.querySelector('.d2l-input-date-hidden-text');
 		this._hiddenContentResizeObserver = new ResizeObserver(() => {
 			const width = Math.ceil(parseFloat(getComputedStyle(hiddenContent).getPropertyValue('width')));
 			this._hiddenContentWidth = `${width}px`;
 		});
 		this._hiddenContentResizeObserver.observe(hiddenContent);
+
+		const hiddenCalendar = this.shadowRoot.querySelector('.d2l-input-date-hidden-calendar');
```

Comment:
This is the same idea as with the hidden text used for calculation of text width. I did choose to remove the calendar after initial render, since if language changes the font-size heights generally seem to be the same and this doesn't seem like it would need to be recalculated.